include/nbl/asset/IAssetManager.h

```diff
@@ -802,13 +802,10 @@ class IAssetManager : public core::IReferenceCounted, public core::QuitSignallin
             return insertAssetIntoCache(_asset, IAsset::EM_IMMUTABLE);
         }
 
-        static inline std::string getFileExt(const io::path& _filename)
+        static inline std::string getFileExt(const std::filesystem::path& _filename)
```

Comment:
this is a duplicate implementation of a function that is already in filesystem

Reply:
> this is a duplicate implementation of a function that is already in filesystem

Not completely. I left it because std::filesystem::path.extension() returns extension in format `.ext`, and getFileExt() returns it it forman 'ext' so it does call std::filesystem::path.extension() but performs additionsl dot removal

Reply:
ok lets make a header `nbl/system/path.h`

```cpp
#ifndef __NBL_SYSTEM_PATH_H_INCLUDED__
#define __NBL_SYSTEM_PATH_H_INCLUDED__

#include <filesystem>

namespace nbl::system
{

using path = std::filesystem::path;

// extra path utils, feel free to add more
inline nbl::string extension_wo_dot(const path& _filename)
{
    return _filename.extension().string().substr(1u,nbl::string::npos);
}

}

#endif
```

remember to add a typedef `nbl::string = std::string` just like we have with the wide strings.

Reply:
Where should i place the typedef?

Reply:
same place where the typedef for wide strings is